### PR TITLE
feat: add transfer speed setting

### PR DESCRIPTION
**Pull request by jjandxa**
# Description
This PR is going to add an transfer setting to the basic preference. The feature request issue is [here](#199 ) and trello task is [here](https://trello.com/c/VXXihpIB/11).
# Screenshots
image

### CODE REVIEW

**Comment by jjandxa** [comment]
sorry, this is miss operation. will going to do recommit.

File: src/renderer/components/Preference/Basic.vue [hunk 1]
```diff
@@ -109,6 +155,8 @@
     const {
       autoCheckUpdate,
       dir,
+      maxOverallUploadLimit,
```
Comment (agalwood) [inline src/renderer/components/Preference/Basic.vue]: Place the keys in alphabetical order.

File: src/renderer/components/Preference/Basic.vue [hunk 2]
```diff
@@ -109,6 +155,8 @@
...
+      maxOverallDownloadLimit,
```
Comment (agalwood) [inline src/renderer/components/Preference/Basic.vue]: Place the keys in alphabetical order.

File: src/renderer/components/Preference/Basic.vue [hunk 3]
```diff
@@ -123,6 +171,8 @@
       autoCheckUpdate,
       continue: config.continue,
       dir,
+      maxOverallUploadLimit,
```
Comment (agalwood) [inline src/renderer/components/Preference/Basic.vue]: Place the keys in alphabetical order.

File: src/renderer/components/Preference/Basic.vue [hunk 4]
```diff
@@ -123,6 +171,8 @@
...
+      maxOverallDownloadLimit,
```
Comment (agalwood) [inline src/renderer/components/Preference/Basic.vue]: Place the keys in alphabetical order.

File: src/renderer/components/Preference/Basic.vue [hunk 5]
```diff
@@ -98,6 +130,20 @@
...
 </template>
+<style>
+  .download-speed-input {
+    width: 50%!important;
+  }
+  .el-input-group .el-select .el-input {
+    width: 60px;
+  }
+  input::-webkit-outer-spin-button,input::-webkit-inner-spin-button {
+    -webkit-appearance: none !important;
```
[the remark this answers is not in the document]
Comment (jjandxa) [reply src/renderer/components/Preference/Basic.vue]: if not override the global number input style. the number input show as 
image

File: src/renderer/components/Preference/Basic.vue [hunk 6]
```diff
@@ -53,6 +53,38 @@
...
+              </el-radio>
+            </el-radio-group>
+          </el-col>
+          <el-col class="form-item-sub" :span="24">
+            {{ $t('preferences.transfer-speed-download') }}
+            <el-radio-group class="el-radio-group-margin-left" v-model="maxOverallDownloadLimitRadio">
+              <el-radio :label="true">{{ $t('preferences.transfer-speed-no-set') }}</el-radio>
+              <el-radio :label="false">
+                {{ $t('preferences.transfer-speed-set') }}
+                <el-input class="download-speed-input" placeholder="请输入内容" type="number" v-model="maxOverallDownloadLimitValue">
```
[the remark this answers is not in the document]
Comment (jjandxa) [reply src/renderer/components/Preference/Basic.vue]: removed.

File: src/renderer/components/Preference/Basic.vue [hunk 7]
```diff
@@ -98,6 +130,20 @@
...
+    width: 50%!important;
+  }
+  .el-input-group .el-select .el-input {
+    width: 60px;
+  }
+  input::-webkit-outer-spin-button,input::-webkit-inner-spin-button {
+    -webkit-appearance: none !important; 
+  }
+  input[type="number"] {
+    -moz-appearance:textfield;
```
[the remark this answers is not in the document]
Comment (jjandxa) [reply src/renderer/components/Preference/Basic.vue]: removed.

File: src/renderer/components/Preference/Basic.vue [hunk 8]
```diff
@@ -98,6 +130,20 @@
     </el-main>
   </el-container>
 </template>
+<style>
+  .download-speed-input {
+    width: 50%!important;
```
[the remark this answers is not in the document]
Comment (jjandxa) [reply src/renderer/components/Preference/Basic.vue]: changed

File: src/renderer/components/Preference/Basic.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (agalwood) [reply src/renderer/components/Preference/Basic.vue]: Try `.download-speed-input::-webkit-outer-spin-button, .download-speed-input::-webkit-inner-spin-button {` ?

[the remark this answers is not in the document]
Comment (jjandxa) [reply src/renderer/components/Preference/Basic.vue]: changed.

File: src/renderer/components/Theme/Default.scss [hunk 9]
```diff
@@ -228,3 +228,9 @@ img {
     height: 100%;
   }
 }
+
+/* Preferences - Basic
+-------------------------- */
+.el-radio-group-margin-left {
```
Comment (agalwood) [inline src/renderer/components/Theme/Default.scss]: The naming of this class name... improved...

Comment (jjandxa) [reply src/renderer/components/Theme/Default.scss]: The class move to Basic.vue and rename class name.

**Pull request merged**